integration-cli/docker_api_images_test.go

```diff
@@ -23,59 +24,53 @@ func (s *DockerSuite) TestApiImagesFilter(c *check.C) {
 		v := url.Values{}
 		v.Set("filter", filter)
 		status, b, err := sockRequest("GET", "/images/json?"+v.Encode(), nil)
-		c.Assert(err, check.IsNil)
-		c.Assert(status, check.Equals, http.StatusOK)
+		c.Assert(err, checker.IsNil)
+		c.Assert(status, checker.Equals, http.StatusOK)
 
 		var images []image
-		if err := json.Unmarshal(b, &images); err != nil {
-			c.Fatal(err)
-		}
+		err = json.Unmarshal(b, &images)
+		c.Assert(err, checker.IsNil)
 
 		return images
 	}
 
 	errMsg := "incorrect number of matches returned"
-	if images := getImages("utest*/*"); len(images[0].RepoTags) != 2 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("utest"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("utest*"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("*5000*/*"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
+	images := getImages("utest*/*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 2, check.Commentf(errMsg))
+
+	images = getImages("utest")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1, check.Commentf(errMsg))
+
+	images = getImages("utest*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1, check.Commentf(errMsg))
+
+	images = getImages("*5000*/*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1, check.Commentf(errMsg))
 }
 
 func (s *DockerSuite) TestApiImagesSaveAndLoad(c *check.C) {
 	testRequires(c, Network)
 	testRequires(c, DaemonIsLinux)
 	out, err := buildImage("saveandload", "FROM hello-world\nENV FOO bar", false)
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, checker.IsNil)
 	id := strings.TrimSpace(out)
 
 	res, body, err := sockRequestRaw("GET", "/images/"+id+"/get", nil, "")
-	c.Assert(err, check.IsNil)
-	c.Assert(res.StatusCode, check.Equals, http.StatusOK)
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
 
 	defer body.Close()
```

Comment:
mmm no, after the error check.
if there's an error the body will be nil and nobody cares, it shoudn't be closed
if there's not an error I see it after the error check and before the status code check, otherwise the call to defer won't be evaluated if the status code check fails